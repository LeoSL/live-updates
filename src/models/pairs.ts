export type TicketPairType = "BTC-USD" | "BTC-CAD" | "ETH-CAD";

export enum TicketPairEnum {
  "BTC-CAD" = "BTCCAD",
  "BTC-USD" = "BTCUSD",
  "ETH-CAD" = "ETHCAD",
}

type Order = {
  price: string;
};

export type Orderbook = {
  asks: Order[];
  bids: Order[];
};

export type PriceAggregate = {
  high: string;
  low: string;
  close: string;
  average: string;
  percentChange: string;
  orderbook: Orderbook;
};

export type MarketDataResponse = {
  displaySymbol: string;
  baseCurrency: string;
  quoteCurrency: string;
  marketDataLast24HourPriceAggregate: PriceAggregate;
};

export type MarketData = {
  marketDataResponse: MarketDataResponse;
};
