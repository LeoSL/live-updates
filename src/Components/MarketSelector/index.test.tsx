import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TicketPairType } from "../../models/pairs";
import { MarketSelector } from ".";

describe("MarketSelector component", () => {
  const ticketPair: TicketPairType = "BTC-USD";
  const markets: TicketPairType[] = ["BTC-USD", "BTC-CAD"];

  const props = {
    ticketPair,
    markets,
    handleChange: () => "click",
  };

  it("renders properly", () => {
    const { container } = render(<MarketSelector {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the select options properly", () => {
    render(<MarketSelector {...props} />);

    const linkElementPair = screen.getByText(ticketPair);

    // Click to open the selector dropdown
    userEvent.click(screen.getByRole("button"));

    const linkElementMarkets = screen.getByText(markets[1]);

    // Check if the initial BTC-USD option is visible
    expect(linkElementPair).toBeInTheDocument();

    // Check if the BTC-CAD option is visible
    expect(linkElementMarkets).toBeInTheDocument();
  });
});
