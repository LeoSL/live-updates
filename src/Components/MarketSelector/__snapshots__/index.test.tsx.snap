// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketSelector component renders properly 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-marketSelect-1"
    >
      <div
        aria-haspopup="listbox"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        role="button"
        tabindex="0"
      >
        BTC-USD
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="BTC-USD"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
